### PR TITLE
Fix for STORE-1223: Add carbon metrics probes to store

### PR DESCRIPTION
**Pull request by tmkasun**
In this PR I have:
- Add metrics probes to 
- Store asset-api
- Store assets-router
- Store pages-router
- Store APIs-router
- Store tenant-router
    --and wrap existing metrics timers with try finally block

### CODE REVIEW

File: apps/publisher/controllers/apis-router.jag [hunk 1]
```diff
@@ -35,44 +35,47 @@ var mapper = function(path) {
...
-            } else {          
-                var proceed = app.execApiHandlers('onApiLoad', request, response, session, page);
-                if (proceed) {
-                    include(path);
+            var file = new File(path);
+            if (file.isExists()) {
+                request.getMappedPath = mapper(path);
+                var username = user ? user.username : null;
+                var isAuthorized = rxtAPI.permissions.hasAppAPIPermission(page,tenantId,username);
+                if(!isAuthorized){
```
Comment (mushthaq33) [inline apps/publisher/controllers/apis-router.jag]: Formatting issue

Comment (tmkasun) [inline apps/publisher/controllers/apis-router.jag]: Thanks @mushthaq33 , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/931b708ba54b43643839e3b48b30a597c4f4e0d8)

File: apps/publisher/controllers/apis-router.jag [hunk 2]
```diff
@@ -35,44 +35,47 @@ var mapper = function(path) {
...
-    //Determine the tenant by the user
-    if (user) {
-        tenantId = user.tenantId;
-    }
-    //Obtain the set of available pages
-    var app = require('rxt').app;
-    var page = getPage(args.suffix);
-    var path = app.getApiEndpointPath(tenantId, page);
-    if (path) {
+try{
```
Comment (splinter) [inline apps/publisher/controllers/apis-router.jag]: Code Formatting issue, line 38

File: jaggery-modules/rxt/module/scripts/asset/asset.js [hunk 3]
```diff
@@ -261,44 +261,46 @@ var asset = {};
...
-        var namespaceAttribute = this.rxtManager.getNamespaceAttribute(this.type);
+        try {
+            metrics.start(this.constructor.name,'create');
+            validateRequiredFeilds(this.type, options , this.rxtManager);
+            var isDefault = false;
+            if ((options.hasOwnProperty(constants.Q_PROP_DEFAULT)) && (options[constants.Q_PROP_DEFAULT] === true)) {
+                delete options[constants.Q_PROP_DEFAULT];
+                isDefault = true;
+            }
+            if(options.attributes.hasOwnProperty(constants.ASSET_PROVIDER)){
```
Comment (splinter) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Formatting issue , line 272

File: jaggery-modules/rxt/module/scripts/asset/asset.js [hunk 4]
```diff
@@ -369,25 +371,27 @@ var asset = {};
...
-        var asset = this.am.get(options.id);
-        if (!this.rxtManager.isGroupingEnabled(this.type)) {
-            log.debug('Omitting grouping step as the groupingEnabled property in the asset configuration has been disabled');
+        try {
+            metrics.start(this.constructor.name,'update');
+            var isDefault = false;
+            if ((options.hasOwnProperty(constants.Q_PROP_DEFAULT)) && (options[constants.Q_PROP_DEFAULT] === true)) {
+                isDefault = true;
+            }
+            if(options.attributes.hasOwnProperty(constants.ASSET_PROVIDER)){
```
Comment (splinter) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Formatting issue, line 380

File: apps/publisher/controllers/apis-router.jag (hunk 2 above)
Comment (tmkasun) [inline apps/publisher/controllers/apis-router.jag]: Thanks @splinter , I have fixed it in [this commit](https://github.com/tmkasun/carbon-store/commit/a5917ea67af4d60979a9f5f59636c7672843645b#diff-cd2251c905a3e28c7b8dcb12d1d03602)

File: jaggery-modules/rxt/module/scripts/asset/asset.js (hunk 3 above)
Comment (tmkasun) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Thanks @splinter , I have fixed it in [this commit](https://github.com/tmkasun/carbon-store/commit/a5917ea67af4d60979a9f5f59636c7672843645b)

File: jaggery-modules/rxt/module/scripts/asset/asset.js (hunk 4 above)
Comment (tmkasun) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Thanks @splinter , I have fixed it in [this commit](https://github.com/tmkasun/carbon-store/commit/a5917ea67af4d60979a9f5f59636c7672843645b)

**Pull request merged**